src/Async/asyncToPromise.js

```diff
@@ -0,0 +1,26 @@
+/** @license ISC License (c) copyright 2018 original and current authors */
+/** @author Dale Francis */
+
+const curry = require('../core/curry')
+const isSameType = require('../core/isSameType')
+const isFunction = require('../core/isFunction')
+
+const Async = require('../core/types').proxy('Async')
+
+const toPromise = m => {
+  if(!isSameType(Async, m)) {
+    throw new TypeError('asyncToPromise: Async or a function returning an Async required')
+  }
+
+  return m.toPromise()
+}
+
+// asyncToPromise :: Async e a -> Promise a e
+// asyncToPromise :: (a -> Async e b) -> a -> Promise b e
+function asyncToPromise(async) {
```

Comment:
probably do not want to use the reserved word `async` here, typically we use `m` for an ADT.

Reply:
updated the siggy